specification/basic_concepts/basic_concepts.tex

```diff
@@ -38,33 +44,106 @@ \section{Main principles}
+Fixed port identifiers must not be used with public unregulated data types,
+since that is likely to cause unresolvable port identifier collisions.
+This restriction must be followed at all times by all parties that use the specification;
+if a private data type definition is ever disclosed to any other party (i.e., a party that did not author it)
+or to the public at large, such data type \emph{must be devoid of a fixed port identifier}.
+Equipment and systems that fail to follow this requirement will not be considered UAVCAN-compliant.
+
+\begin{UAVCANSimpleTable}{Data type taxonomy}{|l|X|X|}
+    & Regulated & Unregulated \\
+    \bfseries{Public}
+    &
+    Standard and third-party definitions.\newline
+    Fixed port ID are allowed; they are called \emph{``regulated port ID''}.
+    &
+    Third-party definitions distributed separately from the UAVCAN specification.\newline
+    Fixed port ID are \emph{not allowed}.
+    \\
+
+    \bfseries{Private}
+    &
+    Nonexistent category.
+    &
+    Third-party definitions that are not available to anyone except the party that authored them.\newline
+    Fixed port ID are permitted as long as the definition is only used by the party that created it;
+    they are called \emph{``unregulated fixed port ID''}.
+    \\
+\end{UAVCANSimpleTable}
+
+Each of the two sets of valid port identifiers (which are subject identifiers and port identifiers) are
+segregated into three categories: application-specific (i.e., non-fixed or unregulated) identifiers,
+regulated non-standard port identifiers (i.e., data type definitions submitted by third parties),
+and standard port identifiers.
+Fixed unregulated port identifiers can be allocated in the same range where application-specific identifiers are.
+The ranges are documented in chapter \ref{sec:application_layer}.
+
+DSDL processing tools are recommended to prohibit unregulated fixed port identifiers by default,
+unless they are explicitly configured otherwise.
+
+Third party authors that desire to release regulated data type definitions or contribute to the standard data
+type set should contact the UAVCAN maintainers for coordination.
+The maintainers will choose unoccupied fixed port identifiers for use with the new definitions, if necessary.
+Since the set of regulated definitions is maintained in a highly centralized manner,
+it can be statically ensured that no identifier collisions will take place within it;
+also, since the identifier ranges used with regulated definitions are segregated from the rest,
+it is also guaranteed that no regulated port ID will ever be in conflict with any other compliant UAVCAN
+node or system.
+
+The motivation for the prohibition of fixed port identifiers in third-party unregulated public data types is
+derived directly from the above:
+since there is no central repository of unregulated definitions,
+collisions would be likely to happen, which would be extremely damaging to the ecosystem.
```

Comment:
The specification should not discuss ecosystems in any normative section. An ecosystem is something that is completely outside of the purview of this specification.